osh/builtin_printf.py

```diff
@@ -207,15 +207,21 @@ def __call__(self, arg_vec):
             try:
               d = int(s)
             except ValueError:
-              # This works around the fact that in the arg recycling case, you have no spid.
-              if word_spid == runtime.NO_SPID:
-                self.errfmt.Print("printf got invalid number %r for this substitution", s,
-                                  span_id=part.type.span_id)
+              if s[0] in '\'"':
+                try:
+                  d = ord(s[1])
+                except IndexError:
+                  d = 0
```

Comment:
I pulled this and ran the tests, and it works!  Comments:

* How about adding test coverage for this `d = 0` case too?  When you just pass a single or double quote?
* The test confused me because you used `\*`.  Maybe choose `a=97` instead?

I'm probably going to change the exceptions to if statements because of the "mycpp" translation, but that is only a little related to this PR.

Reply:
Sure, will do.  I actually started out with `a=97` but tried to be cute with `*=42` -- but clarity trumps cuteness :)

Reply:
* 114e57d6d1109835da06ebceffad58a1065f7c78 -- switch to using "a" character
* c874e2badadf297213961e4905bda1853e78f8d6 -- add spec test for d = 0 case
* a3d47f87f7a10ead084dcc4b73b736a6cdccbead -- avoided adding an instance of exception handling in this MR, at least.